sycl/include/CL/sycl/detail/property_helper.hpp

```diff
@@ -47,7 +47,8 @@ enum PropWithDataKind {
   ImageUseMutex = 2,
   ImageContextBound = 3,
   BufferMemChannel = 4,
-  PropWithDataKindSize = 5
+  AccPropBufferLocation = 5,
+  PropWithDataKindSize = 6
```

Comment:
Shouldn't the new enum be put at the end?

Reply:
The same change was introduced in another PR as well https://github.com/intel/llvm/pull/5604/files#r818076116

I will change the order if there is no compelling reason from Maxim.

Reply:
I found the reason, there is a comparison somewhere where all other enum of the same type cannot exceed this last enum

https://github.com/intel/llvm/blob/54ede407edeb93b7e9334d1e725f48bf981b2965/sycl/include/CL/sycl/detail/property_list_base.hpp#L94

Reply:
This doesn't break ABI compatibility with existing object files which assume `PropWithDataKindSize` has the value `5`?

Reply:
Hi @alexbatashev , I wonder why is this restriction below added in your [change](https://github.com/intel/llvm/commit/f7d073d1c853a10c93acb0376e6f8ca726aa4058)?

https://github.com/intel/llvm/blob/54ede407edeb93b7e9334d1e725f48bf981b2965/sycl/include/CL/sycl/detail/property_list_base.hpp#L94-L97

It seems to say that whenever a new property kind is introduced, `PropWithDataKindSize` will have to change, which potentially breaks ABI when some object assumes it is some older value. Can this restriction be removed, or does no other binary uses this enum?

Reply:
I believe the intention of the check is to prevent other types from impersonating properties, i.e. I cannot go make
```c++
struct ThisIsNotActuallyAProperty : PropertyWithDataBase {
  static constexpr int getKind() {
    return PropWithDataKind::PropWithDataKindSize;
  }
};
```
and then expect it to work as a property.

Incrementing `PropWithDataKindSize` does not break ABI nor API as the only use of it is in a template function in the header. This template function requires the property type as a template parameter, which means for it to be instantiated it must know the property and as such should also know the property kind. As long as you don't increment existing property kinds you are guaranteed that, even if different instantiations know different values for `PropWithDataKindSize` it will only fail for valid cases.

Reply:
Got it, thanks @steffenlarsen :)

Reply:
Resolving this thread given there is no problem.